### ISSUE TITLE
couldn't use "insert" on list

### ISSUE THREAD

**Issue by mcnoat** [spyder-ide/spyder#14427]
## Description

Spyder version: 4.2.0
Python version: 3.8.6
OS: PureOS (based on Debian 10)
### What steps will reproduce the problem?

I tried to click on a list in the variable explorer (the one generated by the tutorial, called 'test_list_tour') and clicked on 'insert'. After inserting a key and a value I got this error message window here and nothing in the variable explorer pane had changed.

### Traceback
```python-traceback
Exception in comms call get_namespace_view:
  File "/home/moritz/anaconda3/envs/samad/lib/python3.8/site-packages/spyder_kernels/comms/commbase.py", line 340, in _handle_remote_call
    return_value = self._remote_callback(
  File "/home/moritz/anaconda3/envs/samad/lib/python3.8/site-packages/spyder_kernels/comms/commbase.py", line 353, in _remote_callback
    return self._remote_call_handlers[call_name](
  File "/home/moritz/anaconda3/envs/samad/lib/python3.8/site-packages/spyder_kernels/console/kernel.py", line 149, in get_namespace_view
    view = make_remote_view(ns, settings, EXCLUDED_NAMES)
  File "/home/moritz/anaconda3/envs/samad/lib/python3.8/site-packages/spyder_kernels/utils/nsview.py", line 717, in make_remote_view
    data = get_remote_data(data, settings, mode='editable',
  File "/home/moritz/anaconda3/envs/samad/lib/python3.8/site-packages/spyder_kernels/utils/nsview.py", line 700, in get_remote_data
    return globalsfilter(
  File "/home/moritz/anaconda3/envs/samad/lib/python3.8/site-packages/spyder_kernels/utils/nsview.py", line 633, in globalsfilter
    (exclude_private and key.startswith('_')) or
AttributeError: 'int' object has no attribute 'startswith'
Exception in comms call get_var_properties:
  File "/home/moritz/anaconda3/envs/samad/lib/python3.8/site-packages/spyder_kernels/comms/commbase.py", line 340, in _handle_remote_call
    return_value = self._remote_callback(
  File "/home/moritz/anaconda3/envs/samad/lib/python3.8/site-packages/spyder_kernels/comms/commbase.py", line 353, in _remote_callback
    return self._remote_call_handlers[call_name](
  File "/home/moritz/anaconda3/envs/samad/lib/python3.8/site-packages/spyder_kernels/console/kernel.py", line 164, in get_var_properties
    data = get_remote_data(ns, settings, mode='editable',
  File "/home/moritz/anaconda3/envs/samad/lib/python3.8/site-packages/spyder_kernels/utils/nsview.py", line 700, in get_remote_data
    return globalsfilter(
  File "/home/moritz/anaconda3/envs/samad/lib/python3.8/site-packages/spyder_kernels/utils/nsview.py", line 633, in globalsfilter
    (exclude_private and key.startswith('_')) or
AttributeError: 'int' object has no attribute 'startswith'
```

## Versions

* Spyder version: 4.2.0 
* Python version: 3.8.6
* Qt version: 5.12.9
* PyQt5 version: 5.12.3
* Operating System: Linux 4.19.0-13-amd64

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0            :  1.4.0 (OK)
chardet >=2.0.0                 :  3.0.4 (OK)
cloudpickle >=0.5.0             :  1.6.0 (OK)
diff_match_patch >=20181111     :  20200713 (OK)
intervaltree >=3.0.2            :  3.0.2 (OK)
IPython >=4.0                   :  7.19.0 (OK)
jedi =0.17.2                    :  0.17.2 (OK)
jsonschema >=3.2.0              :  3.2.0 (OK)
keyring >=17.0.0                :  21.5.0 (OK)
nbconvert >=4.0                 :  6.0.7 (OK)
numpydoc >=0.6.0                :  1.1.0 (OK)
parso =0.7.0                    :  0.7.0 (OK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.7.3 (OK)
pygments >=2.0                  :  2.7.2 (OK)
pylint >=1.0                    :  2.6.0 (OK)
pyls >=0.36.1;<1.0.0            :  0.36.1 (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.1.1             :  0.2.1 (OK)
qdarkstyle >=2.8                :  2.8.1 (OK)
qtawesome >=0.5.7               :  1.0.1 (OK)
qtconsole >=4.7.7               :  5.0.1 (OK)
qtpy >=1.5.0                    :  1.9.0 (OK)
rtree >=0.8.3                   :  0.9.4 (OK)
setuptools >=39.0.0             :  49.6.0.post20201009 (OK)
sphinx >=0.6.6                  :  3.3.1 (OK)
spyder_kernels >=1.10.0;<1.11.0 :  1.10.0 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3               :  0.10.4 (OK)
xdg >=0.26                      :  0.26 (OK)
zmq >=17                        :  20.0.0 (OK)

# Optional:
cython >=0.21                   :  None (NOK)
matplotlib >=2.0.0              :  3.3.3 (OK)
numpy >=1.7                     :  1.19.4 (OK)
pandas >=1.1.1                  :  1.1.4 (OK)
scipy >=0.17.0                  :  1.5.3 (OK)
sympy >=0.7.3                   :  None (NOK)
```

**Comment by mcnoat** [comment]
Update: The problem seemed to have solved itself after restarting Spyder.
Before the restart, I also got an error message window when simply loading pandas in the console - so I don't think that the error was specific to inserting something in the variable explorer.

**Comment by mcnoat** [comment]
Addendum: I just noticed that the first time I started Spyder (i.e. the instance for which I got all those error messages) I got the error message 'Failed to load module "canberra-gtk-module"' several times in the terminal from which I started Spyder. For  the second instance (i.e. the one in which I got no more error messages inside Spyder) the 'Failed to load module' message did not show up in the terminal.
